Event/RouterGenerateEvent.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2014 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Component\Routing\Event;
+
+use Symfony\Component\EventDispatcher\Event;
+
+/**
+ * Event fired before the dynamic router generates a url for a route
+ * The name, parameters and absolute properties are used by the url generator
+ * @see Symfony\Component\Routing\Generator\UrlGeneratorInterface::generate()
+ */
+class RouterGenerateEvent extends Event
+{
+    /**
+     * The route name to generate a url for
+     * 
+     * @var string
+     */
+    private $name;
+
+    /**
+     * The parameters to use when generating the url
+     * 
+     * @var array
+     */
+    private $parameters;
+
+    /**
+     * Whether the generated url should be absolute
+     * 
+     * @var array
+     */
+    private $absolute;
+
+    /**
+     * @param string $name 
+     * @param array $parameters 
+     * @param bool $absolute 
+     */
+    public function __construct($name, $parameters, $absolute)
+    {
+        $this->name = $name;
+        $this->parameters = $parameters;
+        $this->absolute = $absolute;
+    }
+
+    /**
+     * Get route name
+     * 
+     * @return string
+     */
+    public function getName()
```

Comment:
some of the phpdoc on DynamicRouter is wrong - we handle also non-string route "names", for example `Route` objects. see also https://github.com/symfony-cmf/Routing/blob/master/ProviderBasedGenerator.php#L49

we should correctly document this here to make things clear. if you can fix the doc on DynamicRouter::generate as well, that would be nice.